packages/models/src/agreement/agreement.ts

```diff
@@ -56,6 +56,9 @@ export const AgreementStamps = z.object({
 });
 export type AgreementStamps = z.infer<typeof AgreementStamps>;
 
+export const agreementStampKind = AgreementStamps.keyof();
+export type AgreementStampKind = z.infer<typeof agreementStampKind>;
```

Comment:
We usually define the Zod and TS type with the same name and casing! 
The cases where we use the lowercase first letter are when we define the helper object that we use to derive the Zod type (e.g., delegationContractKind), but in this case we don't have it and we are extracting the Zod type directly from another existing type

```suggestion
export const AgreementStampKind = AgreementStamps.keyof();
export type AgreementStampKind = z.infer<typeof AgreementStampKind>;
```

Reply:
@taglioni-r / @shuyec This and the other corresponding comment still need to be addressed